StabilityPackage.wl

```diff
@@ -1,228 +1,459 @@
+evecscale::usage = 
+"finds eigenvectors while scaling the matrix to reasonable numbers"
+ndensities::usage = 
+"finds the neutrino number densities for a species, and returns the matrix of densities in an angular bin"
+esysscale::usage=
+"returns the real and imaginary parts of eigenvalues and associated eigenvectors"
+getInitialGuess::usage=
+"generates Initial Guesses for the first radius from expansion expresssion"
+eboxfitSingleRadius::usage=
+"fit a single radius using the box fit parameters"
+Com::usage=
+"test"
+
+
+
+
+
+ClearAll["StabililtyPackage`", "StabililtyPackage`"]
+
+
+c=2.99792458 10^10; (* cm/s*)
+h=6.6260755 10^-27; (*erg s*)
+hbar =h/(2 Pi); (*erg s*)
+Gf=1.1663787 10^-5; (*GeV^-2*)
+everg=1.60218 10^-12; (* convert eV to ergs*)
+Geverg = everg*10^9; (* convert GeV to ergs *)
+Meverg = everg*10^6; (*convert Mev to erg*) 
+ergev=1.0/everg; (*convert ergs to eV*)
+ergmev=ergev/10^6; (*convert erg to MeV*)
+mp=1.6726219 10^-24; (*Proton mass in g*) 
+munits=Sqrt[2] (Gf/Geverg^2 )(hbar c)^3; (*Sqrt[2] Gf in erg cm^3*)
+\[CapitalDelta]m12sq=(7.59 10^-5) everg^2;
+\[Omega]EMev[En_]:=(\[CapitalDelta]m12sq)/(2 (En Meverg));
+
+Begin["`Private`"]
+
+
+Com[A_,B_]:=Module[{a=A,b=B},Return[A.B-B.A]];
+
+
+
+
+
+	
+(*Have xflavor option  set the x do to 0 here*)
+
+ImportData[infile_]:=
+Association[
+"Endensity"->Import[infile,{"Data","distribution(erg|ccm,lab)"}] (*distribution functions*),
+"matters"->Import[infile,{"Data","rho(g|ccm,com)"}], (*densities*)
+"Yes"->Import[infile,{"Data","Ye"}], (*electron fractions *)
+"freqs"->Import[infile,{"Data", "distribution_frequency_grid(Hz,lab)"}], (*freq grid in hz*)
+"freqmid"->Import[infile,{"Data", "distribution_frequency_mid(Hz,lab)"}], (*freq mid points*)
+"muss"->Import[infile,{"Data", "distribution_costheta_grid(lab)"}], (*Cos\[Theta] grid*)
+"mids"->Import[infile,{"Data", "distribution_costheta_mid(lab)"}], (*Cos\[Theta] bin midpoints*)
+"radius"-> Import[infile,{"Data","r(cm)"}]
+];
+
+
+
+SelectSingleRadius[data_,ri_]:=
+Association[
+"Endensity"->data["Endensity"][[ri]] (*distribution functions*),
+"matters"->data["matters"][[ri]], (*densities*)
+"Yes"->data["Yes"][[ri]], (*electron fractions *)
+"freqs"->data["freqs"], (*freq grid in hz*)
+"freqmid"->data["freqmid"], (*freq mid points*)
+"muss"->data["muss"], (*Cos\[Theta] grid*)
+"mids"->data["mids"] (*Cos\[Theta] bin midpoints*)
+];
+
+
+Options[ndensities]={"xflavor"-> True};
+ndensities[data_,OptionsPattern[]]:=Module[{n,nudensity,nubardensity,nuxdensity,nd,ndb,ndx},
+
+n=Length[data["mids"]];
+
+nudensity[dt_]:= Sum[Sum[data["Endensity"][[1,f,dt,dp]]/ (h (data["freqmid"][[f]])) ,{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+nubardensity[dt_]:= Sum[Sum[data["Endensity"][[2,f,dt,dp]]/ (h (data["freqmid"][[f]])),{f,1,Length[data["freqs"]]-1}],{dp,1,2}];
+
+If[OptionValue["xflavor"],
+nuxdensity[dt_]:= Sum[Sum[0.25 data["Endensity"][[3,f,dt,dp]]/ (h (data["freqmid"][[f]]) ),{f,1,Length[data["freqs"]]-1}],{dp,1,2}]

```

Comment:
Would be good to comment where this 0.25 comes from for when you forget in the future and think "why is this here - all of my results are invalid because I have a wrong number here!" (the x data represents four species)